src/SFA.DAS.AssessorService.Application.Api.External/Models/Search/SearchQuery.cs

```diff
@@ -0,0 +1,63 @@
+﻿using System;
+
+namespace SFA.DAS.AssessorService.Application.Api.External.Models.Search
+{
+    public class SearchQuery : IEquatable<SearchQuery>
+    {
+        public long Uln { get; set; }
+        public string Surname { get; set; }
+        public int UkPrn { get; set; }
+        public string Username { get; set; }
+
+        #region GetHashCode, Equals and IEquatable
+        public override int GetHashCode()
+        {
+            unchecked
+            {
+                const int hashBase = (int)2166136261;
+                const int multiplier = 16777619;
+
+                int hash = hashBase;
+                hash = (hash * multiplier) ^ Uln.GetHashCode();
+                hash = (hash * multiplier) ^ (Surname is null ? 0 : Surname.GetHashCode());
+                hash = (hash * multiplier) ^ UkPrn.GetHashCode();
+                hash = (hash * multiplier) ^ (Username is null ? 0 : Username.GetHashCode());
+                return hash;
+            }
+        }
```

Comment:
Are you using any of this equality stuff? If not, get rid for now until needed. 👍

Reply:
Yeah I'm using it for my Unit Test Asserts . It's similar to what ReSharper generates.